pkg/venafi/tpp/connector.go

```diff
@@ -414,6 +414,38 @@ func (c *Connector) requestMetadataItems(dn string) ([]metadataKeyValueSet, erro
 	return response.Data, err
 }
 
+// Retrieve user's self identity
+func (c *Connector) RetrieveSelfIdentity() (response string, err error) {
+
+	var respIndentities = &IdentitiesResponse{}
+
+	statusCode, statusText, body, err := c.request("GET", urlRetrieveSelfIdentity, nil)
+	if err != nil {
+		log.Printf("Failed to get the used user. Error: %v", err)
+		return "", err
+	}
+	log.Printf("Status code: %d", statusCode)
+
+	switch statusCode {
+	case http.StatusOK:
+		err = json.Unmarshal(body, respIndentities)
+		if err != nil {
+			return "", fmt.Errorf("failed to parse identity response: %s, body: %s", err, body)
+		}
+
+		if (respIndentities != nil) && (len(respIndentities.Identities) > 0) {
+			return respIndentities.Identities[0].Name, nil
+		}
+	case http.StatusUnauthorized:
+		return "", verror.AuthError
+	}
+	return "", fmt.Errorf("failed to get Self. Status code: %d, Status text: %s", statusCode, statusText)
+}
+
+func (c *Connector) RetrieveServiceVersion() (string, error) {
+	return c.requestSystemVersion()
```

Comment:
Why this method is being encapsulated? Won't be better to just rename `requestSystemVersion` to `RequestSystemVersion` so it is exposed outside the package in the connector interface (and by that, to avoid boilerplate code)?

Reply:
@luispresuelVenafi the encapsulation has been removed and the method was renamed instead